pinot-spi/src/main/java/org/apache/pinot/spi/queryeventlistener/BrokerQueryEventInfo.java

```diff
@@ -0,0 +1,464 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.queryeventlistener;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.apache.pinot.spi.trace.RequestContext;
+
+public class BrokerQueryEventInfo {
```

Comment:
What is the difference between this vs `RequestContext`? IMO we should just use `RequestContext` without introducing another layer of info since they serve the same purpose

Reply:
@tibrewalpratik17 : can you also help clarify if we will need a different query event info object for v2?

Reply:
@Jackie-Jiang BrokerQueryEventInfo will have a lot more information which is not there in RequestContext but available in BrokerResponse. I am planning to use getters / setters of those variables to set those values via BrokerResponse. We can't take dependency on BrokerResponse here as it will lead to cyclic dependency. Also we are planning to add a server-level event-listener as well for which we will create `ServerQueryEventInfo` so this class is in parallel with that approach as well.

@ankitsultana we don't need a different query event info object for v2 I think. We can reuse this class info with stageStats variable we get in BrokerResponse.